main.go

```diff
@@ -1,33 +1,35 @@
+/***************************************************************************
+ *
+ * Copyright (c) 2020 Bonc.com.cn, Inc. All Rights Reserved
+ *
+ **************************************************************************/
+
+/**
+ * @file    main_test.go.go
+ * @author  qihouliang(qihouliang@bonc.com.cn)
+ * @date    2023/4/11 22:26
+ * @brief
+ */
+
 package main
 
 import (
-	"context"
-	"os"
-	"os/signal"
-	"syscall"
-
+	"fmt"
 	"github.com/appleboy/CodeGPT/cmd"
+	"github.com/appleboy/CodeGPT/webhook"
+	"github.com/spf13/viper"
+	"log"
+	"net/http"
 )
 
-func withContextFunc(ctx context.Context, f func()) context.Context {
-	ctx, cancel := context.WithCancel(ctx)
-	go func() {
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
-		defer signal.Stop(c)
-
-		select {
-		case <-ctx.Done():
-		case <-c:
-			cancel()
-			f()
-		}
-	}()
-
-	return ctx
-}
-
 func main() {
-	ctx := withContextFunc(context.Background(), func() {})
-	cmd.Execute(ctx)
+	log.SetFlags(log.Lshortfile | log.LstdFlags)
+	cmd.InitConfig()
+	http.HandleFunc("/payload", webhook.HandleWebhook)
+	ip := viper.GetString("server.ip")
+	port := viper.GetString("server.port")
+	var address = fmt.Sprintf("%s:%s", ip, port)
+	fmt.Println("Listening for GitHub Webhooks on", address)
+	http.ListenAndServe(address, nil)
+
 }
```

Comment:
Overall, the code looks clean and well-structured. However, there are a few suggestions for improvement:

1. Error handling: The code should handle errors that might occur during execution, such as errors returned by `http.ListenAndServe()`. Currently, the code simply prints the error to the console and continues execution, which is not a good practice.

2. Logging: The code uses the `fmt.Println()` function to print to the console. It would be better to use a proper logging library, such as `logrus` or `zap`, to provide more control over logging output and formatting.

3. Context usage: The `withContextFunc()` function doesn't seem to serve any purpose in the current code. It creates a context with a cancellation function but doesn't use it anywhere. It could be removed.

4. Unused imports: The code imports the `context` and `syscall` packages, but doesn't use them. These imports should be removed to avoid clutter.

5. Security: The code doesn't seem to have any major security vulnerabilities, but it's always a good practice to sanitize and validate input received from external sources, such as the `webhook.HandleWebhook()` function.